enso/sample/__init__.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+import logging
+from sklearn.metrics.pairwise import pairwise_distances
+
+
+def sample(sampler, data, train_indices, train_size):
+    sampler = Sampler._class_for(sampler)(data, train_indices, train_size)
+    return sampler.sample()
+
+
+class Sampler(object):
+    """Base class for all samples."""
+
+    def __init__(self, data, train_indices, train_size):
+        self.data = data
+        self.train_indices = train_indices
+        self.train_size = train_size
+
+    def sample(self):
+        raise NotImplementedError
+
+    @classmethod
+    def _class_for(cls, sampler_string):
+        for subclass in cls.__subclasses__():
+            if subclass.__name__ == sampler_string:
+                return subclass
+        raise ValueError("Invalid sampler attempted ({})".format(sampler_string))
+
+
+class Random(Sampler):
```

Comment:
Usually put these in named files in the sample directory, with only the base class in init